src/systems/systems/src/hydraulic/mod.rs

```diff
@@ -1622,6 +1625,185 @@ impl SimulationElement for LeakMeasurementValve {
+        //     self.usable_level_modifier().get::<ratio>()
+        // );
+    }
+
+    fn update_forces(&mut self, context: &UpdateContext) -> Vector3<f64> {
+        let gravity_force = context.acceleration_plane_reference_filtered_ms2_vector()
+            * self.virtual_mass.get::<kilogram>();
+
+        let spring_force =
+            self.spring
+                .update_force_nm(context, self.reference_point_cg, self.fluid_cg_position);
+
+        let viscosity_damping = -self
+            .fluid_cg_speed
+            .component_mul(&self.anisotropic_damping_constant);
+
+        gravity_force + spring_force + viscosity_damping
+    }
+
+    fn update_speed_position(&mut self, context: &UpdateContext) {
+        let acceleration = self.update_forces(context) / self.virtual_mass.get::<kilogram>();
+
+        self.fluid_cg_speed += acceleration * context.delta_as_secs_f64();
+
+        self.fluid_cg_position += self.fluid_cg_speed * context.delta_as_secs_f64();
+    }
+
+    // Returns a coefficient of the actual level that will be seen by the gauge.
+    // Example: 0.5 means gauge reads half the actual level of the reservoir
+    fn gauge_modifier(&self) -> Ratio {
+        const LATERAL_BREAKPOINTS: [f64; 6] = [-1., -0.2, 0., 0.2, 0.4, 1.];
+        const LATERAL_MAP: [f64; 6] = [0.2, 0.95, 1., 1.05, 1.2, 1.3];
+
+        const LONGITUDINAL_BREAKPOINTS: [f64; 6] = [-1., -0.1, 0., 0.1, 0.4, 1.];
+        const LONGITUDINAL_MAP: [f64; 6] = [0.2, 0.98, 1., 0.98, 0.2, 0.2];
+
+        const VERTICAL_BREAKPOINTS: [f64; 6] = [-1., -0.1, 0., 0.1, 0.2, 1.];
+        const VERTICAL_MAP: [f64; 6] = [1., 1., 0.7, 0.9, 1.2, 1.5];
+
+        let lateral_ratio = interpolation(
+            &LATERAL_BREAKPOINTS,
+            &LATERAL_MAP,
+            self.fluid_cg_position[0],
+        );
+
+        let logitudinal_ratio = interpolation(
+            &LONGITUDINAL_BREAKPOINTS,
+            &LONGITUDINAL_MAP,
+            self.fluid_cg_position[2],
+        );
+
+        let vertical_ratio = interpolation(
+            &VERTICAL_BREAKPOINTS,
+            &VERTICAL_MAP,
+            self.fluid_cg_position[1],
+        );
+
+        Ratio::new::<ratio>(lateral_ratio * vertical_ratio * logitudinal_ratio)
+    }
+
+    // Returns a coefficient of the actual level that will be available for pumps.
+    // Example: 0.5 means from pumps point of view half the actual level of the reservoir is available
+    fn usable_level_modifier(&self) -> Ratio {
+        const LATERAL_BREAKPOINTS: [f64; 6] = [-1., -0.2, 0., 0.2, 0.4, 1.];
+        const LATERAL_MAP: [f64; 6] = [0.2, 0.8, 1., 1., 1., 1.];
+
+        const LONGITUDINAL_BREAKPOINTS: [f64; 6] = [-1., -0.1, 0., 0.1, 0.4, 1.];
+        const LONGITUDINAL_MAP: [f64; 6] = [0.2, 1., 1., 1., 0.2, 0.2];
+
+        let lateral_ratio = interpolation(
+            &LATERAL_BREAKPOINTS,
+            &LATERAL_MAP,
+            self.fluid_cg_position[0],
+        );
+
+        let logitudinal_ratio = interpolation(
```

Comment:
```suggestion
        let longitudinal_ratio = interpolation(
```

It seems multiline selection in PR reviewing is broken. So I can't make a suggestion which you can immediately commit. Sorry about that 😀.